internal/saucecloud/retry/rdcretrier.go

```diff
@@ -0,0 +1,94 @@
+package retry
+
+import (
+	"context"
+	"fmt"
+	"github.com/rs/zerolog/log"
+	"github.com/saucelabs/saucectl/internal/job"
+	"github.com/saucelabs/saucectl/internal/junit"
+	"github.com/saucelabs/saucectl/internal/msg"
+	"strings"
+)
+
+// testOptionsToCopy indicates, for each framework, which testOptions should be kept untouched
+var testOptionsToCopy = map[string][]string{
+	"espresso": {"numShards", "shardIndex", "clearPackageData", "useTestOrchestrator"},
+}
+
+type RDCRetrier struct {
+	Kind      string
+	RDCReader job.Reader
+}
+
+func getKeysFromMap(mp map[string]bool) []string {
+	keys := make([]string, len(mp))
+
+	i := 0
+	for k := range mp {
+		keys[i] = k
+		i++
+	}
+	return keys
+}
+
+func getFailedClasses(report junit.TestSuites) []string {
+	classes := map[string]bool{}
+
+	for _, s := range report.TestSuites {
+		for _, tc := range s.TestCases {
+			if tc.Error != "" || tc.Failure != "" {
+				classes[tc.ClassName] = true
+			}
+		}
+	}
+	return getKeysFromMap(classes)
+}
+
+func (b *RDCRetrier) keepOnlyMandatoryOptions(testOptions map[string]interface{}) map[string]interface{} {
```

Comment:
I find myself wondering if this function is really necessary? 🤔

Can't we just carry over all of the options and override the `class` one with our list of test classes?

Reply:
After thinking deeper on this, that's not needed. If there is eventually some `notClass`, or `package` or `annotation` they should not disturb anything, that will just scope out more thing, and probably be more precise without the clean-up !